### DISCUSSION TITLE
Ascii or unicode for third party library functions?

### DISCUSSION THREAD

**Discussion by emummel20** [uiua-lang/uiua#476] [General]
[This gentleman](https://www.youtube.com/watch?v=dcWj9zR9SRQ) brings up a good point around 11:15 - What is the expectation or idea for third party libraries? Will folks be able to choose their own glyphs, and then every time you import a new library you have to learn all the glyphs?

Or is the expectation (and this I think would be preferable to me) that glyphs are only really used in the stdlib, and all exported functions will be ascii?

**Comment by kaikalii** [comment]
I don't have any guidance on this at the moment.
While you can certain export functions with glyph names, they are not easy to type.
There are various ways I have thought of to add custom glyph names to the formatted, but they each have there own complications and drawbacks.

**Reply by emummel20** [reply]
Alright, thanks.

My main interest in Uiua is the fact that it's a homoiconic✅, stack-based✅, array lang✅ , none of which really truly _require_ glyphs, though I understand the interest in them.

I just think, while it's great that you'll _be able_ to export glyphs, imho it would not scale well for a user have to learn all new glyphs for _every_ function every time they want to use a new third party lib. Perhaps the general best practice would be to use new glyphs sparingly beyond the stdlib?

**Reply by kaikalii** [reply]
Yes, I think I'd tend to agree.
